main.tf

```diff
@@ -44,11 +46,13 @@ resource "azurerm_network_security_rule" "custom_rules" {
   protocol                                   = lookup(var.custom_rules[count.index], "protocol", "*")
   source_port_ranges                         = split(",", replace(lookup(var.custom_rules[count.index], "source_port_range", "*"), "*", "0-65535"))
   destination_port_ranges                    = split(",", replace(lookup(var.custom_rules[count.index], "destination_port_range", "*"), "*", "0-65535"))
-  source_address_prefix                      = length(lookup(var.custom_rules[count.index], "source_application_security_group_ids", [])) == 0 ? lookup(var.custom_rules[count.index], "source_address_prefix", "*") : ""
-  destination_address_prefix                 = length(lookup(var.custom_rules[count.index], "destination_application_security_group_ids", [])) == 0 ? lookup(var.custom_rules[count.index], "destination_address_prefix", "*") : ""
+  source_address_prefix                      = length(lookup(var.custom_rules[count.index], "source_application_security_group_ids", [])) == 0 && length(lookup(var.custom_rules[count.index], "source_address_prefixes", [])) == 0 ? lookup(var.custom_rules[count.index], "source_address_prefix", "*") : null
+  destination_address_prefix                 = length(lookup(var.custom_rules[count.index], "destination_application_security_group_ids", [])) == 0 && length(lookup(var.custom_rules[count.index], "destination_address_prefixes", [])) == 0 ? lookup(var.custom_rules[count.index], "destination_address_prefix", "*") : null
+  source_address_prefixes                    = length(lookup(var.custom_rules[count.index], "source_application_security_group_ids", [])) == 0 ? lookup(var.custom_rules[count.index], "source_address_prefixes", null) : null
+  destination_address_prefixes               = length(lookup(var.custom_rules[count.index], "destination_application_security_group_ids", [])) == 0 ? lookup(var.custom_rules[count.index], "destination_address_prefixes", null) : null
```

Comment:
Would you mind only using source_address_prefixes in azurerm_network_security_rule instead of the two to simplify the code?

Reply:
Yes, I would mind, since that restricts the possibility of being as specific as possible